integration/src/test/java/org/mule/test/construct/FlowRefTestCase.java

```diff
@@ -342,6 +344,36 @@ public void backpressureFlowRefMaxConcurrencySub() throws Exception {
     assertThat(httpClient.send(request).getStatusCode(), is(SERVICE_UNAVAILABLE.getStatusCode()));
   }
 
+  @Test
+  @Story(BACKPRESSURE)
+  @Issue("MULE-19328")
+  public void backpressureMustNotBeTriggeredAfterReferencedFlowRestart() throws Exception {
+    FlowRunner outerFlow = flowRunner("outerFlow");
+    FlowRunner referencedFlowWithMaxConcurrency = flowRunner("referencedFlowWithMaxConcurrency");
+    outerFlow.dispatchAsync(asyncFlowRunnerScheduler);
+    probe(RECEIVE_TIMEOUT, 50, () -> awaiting.get() == 1);
+    referencedFlowWithMaxConcurrency.restartFlow();
```

Comment:
Inject the flow itself and stop/start it instead of using s flow runner for that (you don't intend to run it!)